File: C++/프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> str_list) {
    vector<string> answer;
    for(int i = 0; i < str_list.size(); i++) {
        if(str_list[i] =="l") {
            answer.assign(str_list.begin(), str_list.begin() + i);
            break;
        } else if (str_list[i] == "r") {
            answer.assign(str_list.begin() + ++i, str_list.end());
            break;
        }
    }
    return answer;
}

File: Python3/프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.py
def solution(str_list):
    l_idx = 21 if str_list.count("l") == 0 else str_list.index("l")
    r_idx = 21 if str_list.count("r") == 0 else str_list.index("r")
    return str_list[:l_idx] if l_idx < r_idx else str_list[r_idx + 1 :]